### PR TITLE
Fixes bug in kill task scheduling DruidCoordinatorSegmentKiller.findIntervalForKillTask()

### PR DESCRIPTION
**Pull request by himanshug**

### CODE REVIEW

**Comment by fjy** [comment]
@himanshug this fails UT

**Comment by himanshug** [comment]
@fjy fixed, my PR was not up-to-date post #2261 merge.

**Comment by fjy** [comment]
:+1:

File: server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentKiller.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        dataSource,
+        new Interval(
+            0,
+            System.currentTimeMillis()
+            - retainDuration
+        ),
+        limit
+    );
+
+    if (unusedSegmentIntervals != null && unusedSegmentIntervals.size() > 0) {
```
Comment (nishantmonu51) [inline server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentKiller.java]: could this be replaced with return JodaUtils.umbrellaInterval(usedSegmentIntervals) ?

Comment (himanshug) [inline server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentKiller.java]: nice tip, changed.

File: server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentKiller.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            log.error(ex, "Failed to submit kill task for dataSource [%s]", dataSource);
+            if (Thread.currentThread().isInterrupted()) {
+              log.warn("skipping kill task scheduling because thread is interrupted.");
+              break;
+            }
+          }
+        }
+      }
+    }
+    return params;
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentKiller.java]: The return contract on params is ill-defined. Some other methods modify the values in the params before returning them. Can you add a method javadoc describing any intended changes to params, or lack of changes?

Comment (himanshug) [inline server/src/main/java/io/druid/server/coordinator/helper/DruidCoordinatorSegmentKiller.java]: ok, added javadoc.

**Comment by drcrallen** [comment]
:+1: after travis

**Comment by nishantmonu51** [comment]
:+1:

**Pull request merged**